### ISSUE TITLE
Issues while running the init-dbs.sh at sunbird-devops #bodhi

### ISSUE THREAD

**Issue by Abhishek-Govula** [project-sunbird/sunbird-commons#108]
While running the _init-dbs.sh_ at [https://github.com/project-sunbird/sunbird-devops/blob/master/Installation.md#step-3-initialize-dbs](https://github.com/project-sunbird/sunbird-devops/blob/master/Installation.md#step-3-initialize-dbs), I have errors. Output file attached.
[init-dbs-output.txt](https://github.com/project-sunbird/sunbird-commons/files/1457310/init-dbs-output.txt)

**Comment by kochhar** [comment]
@Manojvv @endeepak could you please debug and assist in resolving this issue.

**Comment by Manojvv** [comment]
@Abhishek-Govula looks like cassandra is not running on that machine. Please check once.

**Comment by Abhishek-Govula** [comment]
@Manojvv cassandra is running on the system as a service
Below is the output after running _sudo service cassandra status_
● cassandra.service - LSB: distributed storage system for structured data
   Loaded: loaded (/etc/init.d/cassandra; bad; vendor preset: enabled)
   Active: active (exited) since Thu 2017-11-09 15:29:12 IST; 1h 46min ago
     Docs: man:systemd-sysv-generator(8)
  Process: 1031 ExecStop=/etc/init.d/cassandra stop (code=exited, status=0/SUCCESS)
  Process: 1040 ExecStart=/etc/init.d/cassandra start (code=exited, status=0/SUCCESS)

The status is active. I used install-dbs.sh to install cassandra

**Comment by Manojvv** [comment]
@Abhishek-Govula 
can you please share me the output of below two commands
`ps -ef | grep cassandra`
`cqlsh`

**Comment by Abhishek-Govula** [comment]
Output for ```ps -ef | grep cassandra ```
```
$ps -ef | grep cassandra
kmadmin  30698 30685  0 17:49 pts/4    00:00:00 grep --color=auto cassandra
```

##

Output for ```cqlsh```
```
$cqlsh
Connection error: ('Unable to connect to any servers', {'127.0.0.1': error(111, "Tried connecting to [('127.0.0.1', 9042)]. Last error: Connection refused")})
```

**Comment by Abhishek-Govula** [comment]
I had problems with the version of cassandra installed by the automation scripts. It is installing the latest version, which is for some reason not working for Ubuntu 16.04. I have installed the Cassandra v3.11 and got up the service and ```cqlsh``` running. 

After getting cassandra to run, I have got the following error. 
```
@@@@@@@@@ Cassandra data

PLAY [cassandra] ***************************************************************

TASK [Gathering Facts] *********************************************************
ok: [localhost]

TASK [cassandra-cql-update : Copy the templates] *******************************
ok: [localhost] => (item=cassandra.cql)
ok: [localhost] => (item=pageMgmt.csv)
ok: [localhost] => (item=pageSection.csv)
ok: [localhost] => (item=content_service.cql)

TASK [cassandra-cql-update : Run the cassandra.cql command] ********************
fatal: [localhost]: FAILED! => {"changed": true, "cmd": ["cqlsh", "-f", "/tmp/cassandra.cql"], "delta": "0:08:04.787277", "end": "2017-11-09 19:25:39.299086", "failed": true, "msg": "non-zero return code", "rc": 2, "start": "2017-11-09 19:17:34.511809", "stderr": "/tmp/cassandra.cql:602:InvalidRequest: Error from server: code=2200 [Invalid query] message=\"Column tcupdatedat was not found in table user\"", "stderr_lines": ["/tmp/cassandra.cql:602:InvalidRequest: Error from server: code=2200 [Invalid query] message=\"Column tcupdatedat was not found in table user\""], "stdout": "", "stdout_lines": []}
...ignoring

TASK [cassandra-cql-update : Run the pageMgmt.csv command] *********************
changed: [localhost]

TASK [cassandra-cql-update : Run the pageSection.csv file] *********************
changed: [localhost]

TASK [cassandra-cql-update : Run the content_service.cql file] *****************
changed: [localhost]
``` 

Can I ignore this error?

**Comment by Abhishek-Govula** [comment]
Error while initialising the postgres data, below is the error message 
```
PLAY [postgresql-master] *******************************************************

TASK [Gathering Facts] *********************************************************
ok: [localhost]

TASK [postgresql-data-update : Copy the templates] *****************************
ok: [localhost] => (item=tables_postgres.sql)

TASK [postgresql-data-update : Run the postgresql command] *********************
fatal: [localhost]: FAILED! => {"changed": true, "cmd": ["psql", "-h", "127.0.0.1", "-U", "quartz", "-d", "quartz", "-a", "-f", "/tmp/tables_postgres.sql"], "delta": "0:00:00.098415", "end": "2017-11-09 20:44:20.568977", "failed": true, "msg": "non-zero return code", "rc": 2, "start": "2017-11-09 20:44:20.470562", "stderr": "psql: FATAL:  password authentication failed for user \"quartz\"\nFATAL:  password authentication failed for user \"quartz\"", "stderr_lines": ["psql: FATAL:  password authentication failed for user \"quartz\"", "FATAL:  password authentication failed for user \"quartz\""], "stdout": "", "stdout_lines": []}
	to retry, use: --limit @/home/kmadmin/sunbird/sunbird-devops/ansible/postgresql-data-update.retry

PLAY RECAP *********************************************************************
localhost                  : ok=2    changed=0    unreachable=0    failed=1  
```

Do I need to create any extra user for postgres ? #bodhi

**Comment by Manojvv** [comment]
still cassandra is not able to connect. when you run `cqlsh` you should be able to see database console.

for postgresql-data-update:
1) please verify `application_postgres_password`variable is set. 
2) did install-dbs.sh passed for the postgresql installation?
3) make sure postgresql is running.

**Comment by Abhishek-Govula** [comment]
Found the problem while running the ```init-dbs.sh``` for cassandra. Issue raised #110

**Comment by Abhishek-Govula** [comment]
@Manojvv ```install-dbs.sh``` passed successfully for postgres, but I'm facing the same issue as cassandra. The service was running but the connection for ```psql``` was not successful. I have installed ```postgres3.11``` and got the db running.

Fixed the problem with _postgresql-data-update_, the default user was ```quartz```. Since I have set up the databases manually, I had to create a user named ```quartz``` and a database named ```quartz```, then run the _init-dbs.sh_

**Comment by Manojvv** [comment]
@Abhishek-Govula  
 manzarul will look into the issue with `https://github.com/project-sunbird/sunbird-commons/issues/110 `.
Is there any existing issue which you are facing other than Issue:110?

**Comment by Abhishek-Govula** [comment]
@Manojvv There is a problem with the `pg_hba.conf` while running the `install-dbs.sh` for Postgres as well. Should I raise another issue regarding that ?

**Comment by Manojvv** [comment]
@Abhishek-Govula  
I will check the error and fix. If possible please give the error log.  I assume you not blocked because of this issue,  as you installed postgresql manually.

**Comment by Abhishek-Govula** [comment]
@Manojvv I don't have the error log, but the reason is as follows.

The error is occurring when I install the postgres using the `install-dbs.sh`. The script is reading the hba configuration required for postgres from the inventory configuration.
After this, the `pg_hba.conf` file is replaced with a file containing the inventory configuration.

This configuration file is not properly formatted and this is breaking the currently running service.

This was also the reason why the service was showing a healthy status, but the psql command was failing to connect earlier [here](https://github.com/project-sunbird/sunbird-commons/issues/111)

`pg_hba.conf` created by `install-dbs.sh`
```
local all quartz    peer 
local all api_manager_dev    peer 
local all keycloak    peer 
local all all    peer 
host all all 127.0.0.1/32   md5 
host all all ::1/128   md5 
host all all    md5
```
`pg_hba.conf` what I had used (tab separated instead of space - working fine for me)
```
local   all             quartz                                  peer
local   all             keycloak                                peer
local   all             api_manager_dev                         peer
local   all             all                                     peer
host    all             all             127.0.0.1/32          	md5
```

Hope this explains the problem!

**Comment by endeepak** [comment]
The issue wasn't with tab and space separation in the pg_hba.conf. The
issue was due to empty value for swarm_address_space variable in the
inventory group vars. This lead to invalid config in last line of
configuration

We'll have to look into adding validation in the scripts which can surface
this error with helpful hints for fixing it

**Comment by Abhishek-Govula** [comment]
@endeepak `swarm_address_space` not being mentioned in the inventory group vars was the issue why external connections from the app server were being allowed to connect to Postgres. That issue is fixed. Thanks to you!

This issue is rather different. On a working installation of Postgres, if I run the `install-dbs.sh` even with `swarm_address_space` variable being mentioned in the configuration, it is breaking the existing installation. To be precise, the service would show healthy status, but the `5432` port wouldn't be active and `psql` shell also wouldn't work anymore

I traced down the  `pg_hba.conf` issue by since initially connecting to Postgres using postgres user account. I have kept opened the psql shell connected to default postgres db, then after I run the install-dbs.sh, immediately after the config file is replaced, the shell lost the connection.

Also ```local   all             all                                     peer``` was already mentioned in the auto-generated inventory group vars and local connections and all users were already a part of the conf.
